settings.php

```diff
@@ -700,6 +700,48 @@
                 get_string('importvideos_settingenabled', 'block_opencast'),
                 get_string('importvideos_settingenabled_desc', 'block_opencast'), 1));
 
+        // Import videos: Enable import videos within Moodle core course import wizard feature.
+        // This setting applies to both of import modes, therefore hide_if is only limited to importvideosenabled.
+        $importvideossettings->add(
+            new admin_setting_configcheckbox('block_opencast/importvideoscoreenabled',
+                get_string('importvideos_settingcoreenabled', 'block_opencast'),
+                get_string('importvideos_settingcoreenabled_desc', 'block_opencast'), 1));
+        if ($CFG->branch >= 37) { // The hide_if functionality for admin settings is not available before Moodle 3.7.
+            $importvideossettings->hide_if('block_opencast/importvideoscoreenabled',
+                'block_opencast/importvideosenabled', 'notchecked');
+        }
+
+        // Import Video: define modes (ACL Change / Duplicating Events).
+        $importmodechoices = array(
+            'duplication' => get_string('importvideos_settingmodeduplication', 'block_opencast'),
+            'acl' => get_string('importvideos_settingmodeacl', 'block_opencast')
+        );
+
+        // Set default to duplication mode.
+        $select = new admin_setting_configselect('block_opencast/importmode',
+            get_string('importmode', 'block_opencast'),
+            get_string('importmodedesc', 'block_opencast'),
+            'duplication', $importmodechoices);
+
+        $importvideossettings->add($select);
+
+        if ($CFG->branch >= 37) { // The hide_if functionality for admin settings is not available before Moodle 3.7.
+            $importvideossettings->hide_if('block_opencast/importmode',
+                'block_opencast/importvideosenabled', 'notchecked');
+        }
+
+        // Import Videos: Enable manual import videos feature in ACL change mode.
+        $importvideossettings->add(
+            new admin_setting_configcheckbox('block_opencast/importvideosmanualenabledwithacl',
```

Comment:
Why did you create an additional setting for the manual ACL import instead of doing it like the core import where you have one setting for both import modes?
I think it's easier for admins to decide on the Import mode first and then select where the import should be enabled.
My suggestions:

- [ ] Position Import Mode select element at the top of the settings below the general 'importvideosenabled' setting
- [ ] Use the setting 'importvideosmanualenabled' also for the ACL mode (requires changes in importvideosmanager and other places as well)